src/Compilers/VisualBasic/Portable/Scanner/ScannerBuffer.vb

```diff
@@ -95,6 +96,36 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             Return p
         End Function
 
+        Friend Function TryPeek(at As Integer, <Out()> ByRef ch As Char) As Boolean
```

Comment:
@AlekseyTs I'll have a look at this over the next few days, whilst I analyse reducing the number of additional variable usage. A job for my colour in pens, that way I'll see what affect of defaulting  to `ChrW(0)` on failure has.